sql/core/src/main/scala/org/apache/spark/sql/execution/ui/AllExecutionsPage.scala

```diff
@@ -121,65 +122,247 @@ private[ui] class AllExecutionsPage(parent: SQLTab) extends WebUIPage("") with L
+      // New executions should be shown above old executions by default.
+      executionSortColumn == "ID"
+    )
+    val executionPageSize = Option(parameterExecutionPageSize).map(_.toInt).getOrElse(100)
+    val executionPrevPageSize = Option(parameterExecutionPrevPageSize).map(_.toInt).
+      getOrElse(executionPageSize)
+
+    val page: Int = {
+      // If the user has changed to a larger page size, then go to page 1 in order to avoid
+      // IndexOutOfBoundsException.
+      if (executionPageSize <= executionPrevPageSize) {
+        executionPage
+      } else {
+        1
+      }
+    }
+
+    val tableHeaderId = executionTag // "running", "completed" or "failed"
+
+    try {
+      new ExecutionPagedTable(
+        request,
+        parent,
+        executionData,
+        tableHeaderId,
+        executionTag,
+        UIUtils.prependBaseUri(request, parent.basePath),
+        "SQL", // subPath
+        parameterOtherTable,
+        currentTime,
+        pageSize = executionPageSize,
+        sortColumn = executionSortColumn,
+        desc = executionSortDesc,
+        showRunningJobs,
+        showSucceededJobs,
+        showFailedJobs).table(page)
+    } catch {
+      case e@(_: IllegalArgumentException | _: IndexOutOfBoundsException) =>
+        <div class="alert alert-error">
+          <p>Error while rendering execution table:</p>
+          <pre>
+            {Utils.exceptionString(e)}
+          </pre>
+        </div>
+    }
+  }
 }
 
-private[ui] abstract class ExecutionTable(
+
+private[ui] class ExecutionPagedTable(
+    request: HttpServletRequest,
     parent: SQLTab,
-    tableId: String,
+    data: Seq[SQLExecutionUIData],
+    tableHeaderId: String,
+    executionTag: String,
+    basePath: String,
+    subPath: String,
+    parameterOtherTable: Iterable[String],
     currentTime: Long,
-    executionUIDatas: Seq[SQLExecutionUIData],
+    pageSize: Int,
+    sortColumn: String,
+    desc: Boolean,
     showRunningJobs: Boolean,
     showSucceededJobs: Boolean,
-    showFailedJobs: Boolean) {
+    showFailedJobs: Boolean) extends PagedTable[ExecutionTableRowData] {
 
-  protected def baseHeader: Seq[String] = Seq(
-    "ID",
-    "Description",
-    "Submitted",
-    "Duration")
+  override val dataSource = new ExecutionDataSource(
+    request,
+    parent,
+    data,
+    basePath,
+    currentTime,
+    pageSize,
+    sortColumn,
+    desc)
 
-  protected def header: Seq[String]
+  val parameterPath = basePath + s"/$subPath/?" + parameterOtherTable.mkString("&")
 
-  protected def row(
-      request: HttpServletRequest,
-      currentTime: Long,
-      executionUIData: SQLExecutionUIData): Seq[Node] = {
-    val submissionTime = executionUIData.submissionTime
-    val duration = executionUIData.completionTime.map(_.getTime()).getOrElse(currentTime) -
-      submissionTime
+  override def tableId: String = executionTag + "-table"
```

Comment:
Here and below, how about interpolation consistently?

Reply:
Done. Thanks